pkg/dockerregistry/server/manifestservice.go

```diff
@@ -152,15 +153,26 @@ func (m *manifestService) Put(ctx context.Context, manifest distribution.Manifes
 		}
 	}
 
-	if err = mh.FillImageMetadata(ctx, &ism.Image); err != nil {
+	// FIXME: We have to switch the helpers to use external versions to avoid
+	// these conversions which might be expensive. Since we call bunch of
+	// openshift helpers here, we need to fix this when we switch those to
+	// external versions.
+	image := imageapi.Image{}
+	if err := imageapiv1.Convert_v1_Image_To_image_Image(&ism.Image, &image, nil); err != nil {
```

Comment:
@smarterclayton this is controversial... the FillImageMetadata is pretty heavy helper.